sympy/core/tests/test_equation.py

```diff
@@ -0,0 +1,122 @@
+    assert c ** tsteqn == Equation(c ** a, c ** (b / c))
+    assert tsteqn + tsteqn == Equation(2 * a, 2 * b / c)
+    assert tsteqn * tsteqn == Equation(a ** 2, b ** 2 / c ** 2)
+    assert tsteqn - tsteqn == Equation(0, 0)
+    assert tsteqn / tsteqn == Equation(1, 1)
+    assert tsteqn % tsteqn == Equation(0, 0)
+    assert tsteqn ** tsteqn == Equation(a ** a, (b / c) ** (b / c))
+
+
+def test_outputs():
+    a, b, c = var('a b c')
+    tsteqn = Eqn(a, b / c)
+    assert str(tsteqn) == 'a = b/c'
+    assert latex(tsteqn) == 'a = \\frac{b}{c}'
+
+
+def test_helper_functions():
+    a, b, c = var('a b c')
+    tsteqn = Eqn(a, b / c)
+    raises(ValueError, lambda: integrate(tsteqn, c))
+    raises(AttributeError, lambda: integrate(tsteqn, c, side='right'))
+    assert tsteqn.evalf(4, {b: 2.0, c: 4}) == Eqn(a, 0.5000)
+    assert diff(tsteqn, c) == Equation(diff(a, c, evaluate=False), -b / c ** 2)
+    tsteqn = Eqn(a * c, b / c)
+    assert diff(tsteqn, c) == Equation(a, -b / c ** 2)
+    assert integrate(tsteqn, c, side='rhs') == integrate(tsteqn.rhs, c)
+    assert integrate(tsteqn, c, side='lhs') == integrate(tsteqn.lhs, c)
+
+    def adsq(expr):
+        # Arbitrary python function
+        return expr + expr ** 2
+
+    assert (adsq(tsteqn) == Eqn(a ** 2 * c ** 2 + a * c, b ** 2 / c ** 2 +
+                                b / c))
+    tsteqn = Eqn((a - 1) * (a + 1) * (a + 3), (a + 3) * (2 * b + c) ** 2)
+    tsteqn2 = Eqn(a ** 3 + 3 * a ** 2 - a - 3,
+                  4 * a * b ** 2 + 4 * a * b * c + a * c ** 2 + 12 * b ** 2 +
+                  12 * b * c + 3 * c ** 2)
+    tsteqn3 = Eqn(a ** 3 + 3 * a ** 2 - a - 3,
+                  a * c ** 2 + b ** 2 * (4 * a + 12) + b * (
+                      4 * a * c + 12 * c) + 3 * c ** 2)
+    tsteqn4 = Eqn(a ** 3 + 3 * a ** 2 - a - 3,
+                  a * c ** 2 + 4 * b ** 2 * (a + 3) + 4 * b * c * (
+                      a + 3) + 3 * c ** 2)
+    assert tsteqn.expand() == tsteqn2
+    assert expand(tsteqn) == tsteqn2
+    assert tsteqn2.collect(b) == tsteqn3
+    assert collect(tsteqn2, b) == tsteqn3
+    assert tsteqn3.simplify() == tsteqn4
+    assert simplify(tsteqn3) == tsteqn4
+    assert tsteqn4.factor() == tsteqn
+    assert factor(tsteqn4) == tsteqn
+    # Check matrix exponentiation is not overridden.
+    tsteqn5 = Eqn(a, Matrix([[1, 1], [1, 1]]))
+    assert (exp(tsteqn5).lhs == exp(a))
+    assert (exp(tsteqn5).rhs == exp(Matrix([[1, 1], [1, 1]])))
+
+
+def test_apply_syntax():
+    a, b, c, x = var('a b c x')
+    tsteqn = Eqn(a, b / c)
+    assert tsteqn.apply(log) == Equation(log(a), log(b / c))
+    assert tsteqn.applylhs(log) == Equation(log(a), b / c)
+    assert tsteqn.applyrhs(log) == Equation(a, log(b / c))
+    poly = Eqn(a * x ** 2 + b * x + c * x ** 2, a * x ** 3 + b * x ** 3 + c
+               * x)
+    assert poly.applyrhs(collect, x) == Eqn(poly.lhs, poly.rhs.collect(x))
+
+
+def test_do_syntax():
+    a, b, c, x = var('a b c x')
```

Comment:
Use `symbols` rather than `var` in both library and test code.

Reply:
Done
Commit 4dc62ee7